src/components/README.md

```diff
@@ -63,3 +63,81 @@ import { EntityField } from "@yext/visual-editor";
 ```
+
+## Entity Field Selection
+
+### YextEntityFieldSelector
+
+Use this to allow Visual Editor users to choose an entity field or a constant value that will populate data into a component.
+
+#### Props
+
+| Name                      | Type            | Description                                                                                                          |
+| ------------------------- | --------------- | -------------------------------------------------------------------------------------------------------------------- |
+| useDocument               | function        | See [@yext/pages useDocument](https://github.com/yext/pages/blob/main/packages/pages/src/util/README.md#usedocument) |
+| label?                    | string          | The user-facing label for the field.                                                                                 |
+| filter?.types?            | string[]        | Determines which fields will be available based on field type.                                                       |
+| filter?.includeSubfields? | boolean         |                                                                                                                      |
+| filter?.allowList?        | types: string[] | Field names to include. Cannot be combined with disallowList.                                                        |
+| filter?.disallowList?     | types: string[] | Field names to exclude. Cannot be combined with allowList.                                                           |
+
+### resolveDataForEntityField
+
+Determines whether to use a constant value or an entity field. Used as [Puck's resolveData function](https://puckeditor.com/docs/api-reference/configuration/component-config#resolvedatadata-params).
+
+### Usage
+
+```tsx
+import {
+  EntityFieldType,
+  YextEntityFieldSelector,
+  resolveDataForEntityField,
+} from "@yext/visual-editor";
+import { config } from "../templates/myTemplate";
+import { useDocument } from "@yext/pages/util";
+
+export type ExampleProps = {
+  myField: {
+    entityField: EntityFieldType;
+  };
+};
+
+const exampleFields: Fields<ExampleProps> = {
+  myField: {
+    type: "object",
+    label: "Example Field Populated by Entity Fields",
+    objectFields: {
+      //@ts-expect-error ts(2322)
+      entityField: YextEntityFieldSelector<typeof config>({
+        useDocument: useDocument,
+        label: "Entity Field",
+        filter: {
+          types: ["type.string"],
+          includeSubfields: true,
+          disallowList: ["exampleFieldName"],
+        },
+      }),
+    },
+  },
+};
+
+export const ExampleComponent: ComponentConfig<ExampleProps> = {
+  fields: exampleFields,
+  defaultProps: {
+    myField: {
+      entityField: {
+        name: "",
+        value: "Information",
+      },
+    },
+  },
+  label: "Example Component",
+  resolveData: (props, changed) =>
+    resolveDataForEntityField<ExampleProps>("myField", props, changed),
```

Comment:
Yes -- it's `const resolveDataForEntityField = <T extends DefaultComponentProps>(
  fieldName: keyof T, ...` so it would force it / autocomplete it to a key of ExampleProps here